src/actions/pause/paused.js

```diff
@@ -0,0 +1,55 @@
+// @flow
+
+import {
+  getHiddenBreakpointLocation,
+  isEvaluatingExpression
+} from "../../selectors";
+import { updateFrameLocations } from "../../utils/pause";
+import { removeBreakpoint } from "../breakpoints";
+import { evaluateExpressions } from "../expressions";
+import { selectSource } from "../sources";
+import { mapScopes } from "./mapScopes";
+
+import type { Pause } from "../../types";
+import type { ThunkArgs } from "../types";
+
+/**
+ * Debugger has just paused
+ *
+ * @param {object} pauseInfo
+ * @memberof actions/pause
+ * @static
+ */
+export function paused(pauseInfo: Pause) {
+  return async function({ dispatch, getState, client, sourceMaps }: ThunkArgs) {
```

Comment:
do we need to use async here? or can we make this dispatch utilize thunk, and put the async (mapped frames, and frameScopes) stuff in the [PROMISE] ?